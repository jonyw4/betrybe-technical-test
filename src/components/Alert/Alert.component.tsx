import * as React from 'react';

/**
 * Alert box
 */
export function Alert({ children, type = 'default' }: AlertProps) {
  return <div className={`alert alert--${type}`}>{children}</div>;
}

interface AlertProps {
  children: string;
  type?: 'default' | 'warning';
}
