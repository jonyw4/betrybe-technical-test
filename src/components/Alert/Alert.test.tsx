import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from './Alert.component';

describe('components > Alert', () => {
  it('should renders a default alert correctly', () => {
    const tree = renderer.create(<Alert>Alerta Padr√£o</Alert>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders a warning alert correctly', () => {
    const tree = renderer
      .create(<Alert type="warning">Alerta de Aviso</Alert>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
