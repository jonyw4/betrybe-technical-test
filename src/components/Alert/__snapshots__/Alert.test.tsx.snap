// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Alert should renders a default alert correctly 1`] = `
<div
  className="alert alert--default"
>
  Alerta Padr√£o
</div>
`;

exports[`components > Alert should renders a warning alert correctly 1`] = `
<div
  className="alert alert--warning"
>
  Alerta de Aviso
</div>
`;
